crates/sargon/src/profile/mfa/security_structures/roles/builder/roles_builder.rs

```diff
@@ -539,14 +608,40 @@ impl<const ROLE: u8> RoleBuilder<ROLE> {
         &self,
         factor_source_id: &FactorSourceID,
         factor_list_kind: FactorListKind,
+        mode: SecurityShieldBuilderMode,
     ) -> RoleBuilderMutateResult {
-        if self.contains_factor_source(factor_source_id) {
-            return RoleBuilderMutateResult::forever_invalid(
-                FactorSourceAlreadyPresent,
-            );
+        let duplicates_err = RoleBuilderMutateResult::forever_invalid(
+            FactorSourceAlreadyPresent,
+        );
+
+        match mode {
+            SecurityShieldBuilderMode::Strict => {
+                if self.contains_factor_source(factor_source_id) {
+                    return duplicates_err;
+                }
+            }
+            SecurityShieldBuilderMode::Lenient => {
+                match factor_list_kind {
+                    Override => {
+                        if self
+                            .override_contains_factor_source(factor_source_id)
+                        {
+                            return duplicates_err;
+                        }
+                        // but if threshold contains it, we're good (since mode is lenient)
```

Comment:
With Lenient, we allow a factor which **is** in `override` to be added to `threshold` (and vice versa)